src/cart/crt_ctl.c

```diff
@@ -75,6 +75,113 @@ static int verify_ctl_in_args(struct crt_ctl_ep_ls_in *in_args)
 	return rc;
 }
 
+static int
+crt_ctl_fill_buffer_cb(d_list_t *rlink, void *arg)
+{
+	struct crt_uri_item	*ui;
+	int			*idx;
+	struct crt_uri_cache	*uri_cache = arg;
+	int			 i;
+	int			 rc = 0;
+
+	D_ASSERT(rlink != NULL);
+	D_ASSERT(arg != NULL);
+
+	ui = crt_ui_link2ptr(rlink);
+
+	D_MUTEX_LOCK(&ui->ui_mutex);
+
+	idx = &uri_cache->idx;
```

Comment:
It would be a lot cleaner if you just used udi_cache->idx directly here, L104 in particular looks very weird.